app/controllers/manage_metadata_controller.rb

```diff
@@ -87,148 +87,61 @@ def compare_resource_umm_version(concept_id = nil)
+    # Set an instance variable as setup for the show and revisions page
+    published_resource_name unless params[:controller] == 'collection_associations'
 
-    set_service_information
-  end
-
-  def set_service_information
-    # search for service by concept id to get the native_id and provider_id
-    # if the service is not found, try again because CMR might be a little slow to index if it is a newly published record
-    attempts = 0
-    while attempts < 20
-      services_search_response = cmr_client.get_services(concept_id: @concept_id, all_revisions: true, sort_key: '-revision_date')
-
-      service_data = if services_search_response.success?
-                       services_search_response.body.fetch('items', [])
-                     else
-                       []
-                     end
-      service_data.sort! { |a, b| b['meta']['revision-id'] <=> a['meta']['revision-id'] }
-
-      @revisions = service_data
-      latest = service_data.first
-      meta = latest.nil? ? {} : latest.fetch('meta', {})
-
-      @old_revision = !@revision_id.nil? && meta['revision-id'].to_s != @revision_id.to_s ? true : false
-
-      break if latest && !@revision_id
-      break if latest && meta.fetch('revision-id', 0) >= @revision_id.to_i && meta['concept-id'] == @concept_id
-      attempts += 1
-      sleep 0.05
-    end
-
-    if latest.blank?
-      Rails.logger.error("Error searching for Service #{@concept_id} in `set_service_information`: #{services_search_response.clean_inspect}")
-    else
-      @provider_id = meta['provider-id']
-      @native_id = meta['native-id']
-      @num_associated_collections = cmr_client.get_collections_by_post({ service_concept_id: @concept_id }, token).body.fetch('items', []).size
-    end
-  end
-
-  # TODO: look into refactoring these repetitive methods for V/S/T with
-  # metaprogramming and add sufficient comments to explain
-  def set_tool
-    @concept_id = params[:tool_id] || params[:id]
-    @revision_id = params[:revision_id]
-
-    # retrieve the variable metadata with the current umm_t version
-    headers = { 'Accept' => "application/#{Rails.configuration.umm_t_version}; charset=utf-8" }
-    tool_concept_response = cmr_client.get_concept(@concept_id, token, headers, @revision_id)
-
-    @tool = if tool_concept_response.success?
-              tool_concept_response.body
-            else
-              Rails.logger.error("Error retrieving concept for Tool #{@concept_id} in `set_tool`: #{tool_concept_response.clean_inspect}")
-              {}
-            end
-            
-    set_tool_information
+    set_metadata_information
   end
 
-  def set_tool_information
-    # search for tool by concept id to get the native_id and provider_id
-    # if the tool is not found, try again because CMR might be a little slow to index if it is a newly published record
+  def set_metadata_information
+    # search for metadata record by concept id to get the native_id and provider_id
```

Comment:
for future devs, do we want to say here that the retrying was recommended by CMR?